dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/MetadataBitstreamRestRepository.java

```diff
@@ -220,6 +240,51 @@ private List<FileInfo> getFilePreviewContent(Context context, Bitstream bitstrea
         return fileInfos;
     }
 
+    /**
+     * Create sub map for preview content and file info.
+     * @param sourceMap  parent sub map
+     * @param creator    creator function
+     * @return    created sub map
+     */
+    private <T, U> Hashtable<String, T> createSubMap(Map<String, U> sourceMap, Function<U, T> creator) {
+        Hashtable<String, T> sub = new Hashtable<>();
+        if (sourceMap != null) {
+            for (Map.Entry<String, U> entry : sourceMap.entrySet()) {
+                sub.put(entry.getKey(), creator.apply(entry.getValue()));
+            }
+        }
+        return sub;
+    }
+
+    /**
+     * Create file info from preview content.
+     * @param pc  preview content
+     * @return    created file info
+     */
+    private FileInfo createFileInfo(PreviewContent pc) {
+        Hashtable<String, FileInfo> sub = createSubMap(pc.sub, this::createFileInfo);
+        return new FileInfo(pc.name, pc.content, pc.size, pc.isDirectory, sub);
+    }
+
+    /**
+     * Create preview content from file info for bitstream.
+     * @param context   DSpace context object
+     * @param bitstream bitstream
+     * @param fi        file info
+     * @return          created preview content
+     * @throws SQLException If database error is occurred
+     */
+    private PreviewContent createPreviewContent(Context context, Bitstream bitstream, FileInfo fi) throws SQLException {
+        Hashtable<String, PreviewContent> sub = createSubMap(fi.sub, value -> {
+            try {
+                return createPreviewContent(context, bitstream, value);
+            } catch (SQLException e) {
+                throw new RuntimeException(e);
```

Comment:
Please specify what is happening when this error occurs, just add there some hint that we are trying to create a preview content and please add this errror message also into log.error()